sphinx/source/developers/consensus.rst

```diff
@@ -0,0 +1,41 @@
+Consensus Protocols
+===================
+
+In order to guarantee system reliability changes to the key-value store must be agreed by a number of nodes before being applied. The number of nodes that need to agree on a change before it is applied depends on the consensus protocol being run.
+
+The default consensus protocol for CCF is a version of Raft which provides Crash Fault Tolerance.
+
+We are currently working on providing PBFT as an additional consensus option. PBFT provides Byzantine Fault Tolerance.
+
+.. warning:: Currently CCF with PBFT is in development and should not be used in a production environment.
+
+Raft Consensus Protocol
+-----------------------
+
+We have implemented a version of Raft that provides Crash Fault Tolerance. This means that if we can tolerate up to ``f`` nodes crashing (or being un-responsive) then we need ``2*f + 1`` nodes in total to guarantee liveness. The majority of nodes that need to agree on a change is then ``f + 1``.
+Raft does not make any confidentiality or integrity claims if more than ``f`` nodes have been compromised.
```

Comment:
This is misleading, it implies Raft is robust in some ways to fewer than f nodes being malicious. It is not, if even one node is malicious, then it's possible for it to become leader and issue transactions that are illegal from the state machine's point of view. Raft offers no protection against, regardless of the number of malicious nodes.

Neither Raft nor PBFT offer any confidentiality guarantees, it's also misleading to mention it in conjunction with consensus, we should leave it out. CCF encrypts node to node traffic, and SGX encrypts memory, and that's where confidentiality comes from, our choice of consensus is orthogonal.

Reply:
Yes I just fixed the bit implying that raft can take up to f malicious nodes.

I'll remove the mentions to confidentiality